### PR TITLE
feat: add trace

### PR DESCRIPTION
**Pull request by welkeyever**
#### What type of PR is this?
feat

#### What this PR does / why we need it (English/Chinese):
add trace stub in http2, including: 
* start
* start read header
* finish read header
* start handler
* finish handler
* finish

#### Which issue(s) this PR fixes:
None

### CODE REVIEW

File: trace.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	}
+	if err != nil {
+		ti.Stats().Record(event, stats.StatusError, err.Error())
+	} else {
+		ti.Stats().Record(event, stats.StatusInfo, "")
+	}
+}
+
+// CalcEventCostUs calculates the duration between start and end and returns in microsecond.
+func CalcEventCostUs(start, end traceinfo.Event) uint64 {
```
Comment (Duslia) [inline trace.go]: where use it?

File: server.go [hunk 2]
```diff
@@ -1875,28 +1907,33 @@ func (sc *serverConn) runHandler(rw *responseWriter, reqCtx *app.RequestContext,
 				const size = 64 << 10
 				buf := make([]byte, size)
 				buf = buf[:runtime.Stack(buf, false)]
-				hlog.SystemLogger().Errorf("HTTP2: panic serving %v: %v\n%s", sc.conn.RemoteAddr(), e, buf)
+				err = fmt.Errorf("HTTP2: panic serving %v: %v\n%s", sc.conn.RemoteAddr(), e, buf)
+				hlog.SystemLogger().Error(err.Error())
```
Comment (Duslia) [inline server.go:1911]: It seems that trace doesn't record the error?

File: trace.go (hunk 1 above)
Comment (welkeyever) [reply trace.go]: deleted

File: server.go (hunk 2 above)
Comment (welkeyever) [reply server.go:1911]: Captured by closure.

**Review by Duslia — APPROVED** [review]

**Pull request merged**